src/DynamoCore/Extensions/ExtensionLoader.cs

```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
         /// An event that is raised when an extension starts loading.
         /// </summary>
         public event Action<IExtension> ExtensionLoading;
+
+        internal List<string> directoriesToVerifyCertificates = new List<string>();
+
+        private static bool CheckExtensionCertificates(ExtensionDefinition viewExtension)
+        {
+            //Verify the node library exists in the package bin directory
+            if (!File.Exists(viewExtension.AssemblyPath))
+            {
+                throw new Exception(String.Format(
+                    "An extension called {0} found at {1} is missing dlls which are defined in the view extension definition.  Ignoring it.",
+                    viewExtension.TypeName, viewExtension.AssemblyPath));
+            }
+
+            //Verify that you can load the node library assembly into a Reflection only context
+            Assembly asm;
+            try
+            {
+                asm = Assembly.ReflectionOnlyLoadFrom(viewExtension.AssemblyPath);
+            }
+            catch
+            {
+                throw new Exception(String.Format(
+                    "An extension called {0} found at {1} has a dll which could not be loaded.  Ignoring it.",
+                    viewExtension.TypeName, viewExtension.AssemblyPath));
+            }
+
+            //Verify teh node libarary has a verified signed certificate
+            var cert = asm.Modules.FirstOrDefault()?.GetSignerCertificate();
+            if (cert != null)
+            {
+                var cert2 = new System.Security.Cryptography.X509Certificates.X509Certificate2(cert);
```

Comment:
have you verified this works on linux/mac under mono? .net Core? - if not I wonder if this flow should be disabled if running on non windows platform?